Src/Optimization/PiyavskogoMethod.cpp

```diff
@@ -10,7 +10,7 @@ PiyavskogoMethod::PiyavskogoMethod(double _r)
 }
 
 
-double PiyavskogoMethod::solve(Function& f, double a, double b, double epsilon, int n, std::vector<Test>& history, int& operationCount)
+double PiyavskogoMethod::solve(IFunction& f, double a, double b, double epsilon, int n, std::vector<Test>& history, int& operationCount)

```

Comment:
Not there is no functions that can be passed here in the whole project. Could you also inherit Function from IFunction?

Reply:
@Zebraside, fixed the issues. Please, review.

Reply:
@Zebraside, fixed the comments. Please, review.